In [ ]:
import pandas as pd
import numpy as np

In [ ]:
path = './data.csv'
df = pd.read_csv(path)

### 1. Write a summary for your data
 



### 2. Write Column descriptions

- Timestamp : not needed  [ Numerical Data] 
    - S No : not needed  [ Numerical Data] 
    - Age : Mostly consists Middle aged public data [ Numerical Data] 
    - Gender : Considers Male and Female only [Categorical Data]
    - Country : Mostly United States Citizen data is considered [Categorical Data]
    - state : No missing values [Categorical Data]
    - self_employed : 
                         whether person is self employed or not
                         contains NA, Yes and No values [Categorical Data]
    - family_history:
                         Mention of whether the person's family has any history
                          of mental health issues. Consists Yes and No values.

    - treatment :
                          Whether the person is actively getting treated for
                           the issue he/she has beem facing. 
                           Consists Yes and No values.   

    - work_interfere:  
                         Whether the mental health issues are interfering with 
                         work or not. Answered in Four Values: Often, rarely, 
                         Never,Sometimes and NA .

    - no_employees:
                        Number of employees in the Company or Organisation the
                        person is working in. Consists values in NA, More than
                         1000, 26-100, 100-500 and 500-1000.

    - remote_work:      Whether the person is working from home or not.
                        Consists values in Yes and No. [ Categorical 
                        Data] 

   - tech_company : 
                        Whether the person is working in a tech company or not
                        Consists values in Yes and No 
                        [Categorical Data]
                        
   - benefits :
                       Whether the person is availing any benefits from the 
                       company regarding mental health. Consists values in Yes
                        , No and Don't Know Values.
                       [Categorical Data]                     

   - care_options : 
                      If the company or organization is providing care options
                       for the employee. Consists values in Yes , No and 
                       Not sure values.
                       [Categorical Data]

   - wellness_program :
                       Whether the company or organisation working in conducts 
                       any wellness program or not.
                       Consists values like Yes, No and Don't know.
                       [Categorical Data]

   - seek_help :
                      Whether the person is taking or seeking help from any 
                      person or organisation.
                      Consists values like Yes , No and Don't know.
                      [Categorical Data]

   - anonymity : 
                      Whether the person is anonymously dealing with it 
                      Consists values in Yes, No and Don't know.
                      [Categorical Data]

   - leave : 
                      If the person is able to take leaves for health issues 
                      in his or her company.
                      Consists values in Yes , No and Maybe
                      [Categorical Data]

   - mental_health_consequence :
                     
                      If the work at the organisation is affecting the mental 
                      health of the employee
                      Consists values in Yes , No and Maybe.
                      [Categorical Data]

   - phys_health_consequence :
                     
                      If the work at the organisation is affecting the physical 
                      health of the employee
                      Consists values in Yes , No and Maybe.
                      [Categorical Data]

   - coworkers :
                      Whether the person is being affected mentally because of 
                      coworkers or not.
                      Consists values in Yes, No and Some of them values.
                      [Categorical Data]

   - supervisor :      
                      Whether the person's mental health is being affected by 
                      the Supervisor or higher up officer.
                      Consists values in Yes, No and Some of them values.
                      [Categorical Data]

   - mental_health_interviews :
                     If the person is attending any mental health interviews as 
                     part of treatment.
                     Consists values in Yes , No and Maybe values.
                     [Categorical Data]
  
   - phys_health_interviews :
                     If the person is attending any physical health interviews 
                     as part of treatment.
                     Consists values in Yes , No and Maybe values.
                     [Categorical Data]

   - mental_vs_physical :

  - obs_consequence :
                     If the person had observed any serious consequences due to
                     mental health illness in his/her life.
                     Consists values in Yes and No.
                     [Categorical Data]

  - comments :
                    If the person wants to add any comments about their mental
                    health.


### Issues with the dataset

1. Dirty Data

  Table - `df`

    








### Types of Assessment
There are 2 types of assessment styles

- `Manual` - Looking through the data manually in google sheets or excel
- `Programmatic` - By using pandas functions such as info(), describe() or sample()

### Steps in Assessment
There are 2 steps involved in Assessment

- Discover
- Document

### Issues with the dataset (from manual observation)

 Dirty Data

  Table - `df`

    - `timestamp` column should be dropped
    - `comments` column should be dropped
    - `sno` column should be dropped
    - someone has a negative age in `age` column
    - state', 'self_employed', 'family_history',
       'treatment', 'work_interfere', 'no_employees' columns have missing values
    



### Automatic Assessment

- head and tail
- sample
- info
- isnull
- duplicated
- describe

In [ ]:
df.to_excel("mental.xlsx", index = False)

# DEFINE PROBLEM -> CODE -> TEST

In [ ]:
# DROPPING COLUMNS
df = df.drop(columns=['Timestamp', 'S No', 'comments'])

In [ ]:
df.head()

Age Gender         Country state self_employed family_history treatment  \
0   37      F   United States    IL           NaN             No       Yes   
1   44      M   United States    IN           NaN             No        No   
2   32      M          Canada   NaN           NaN             No        No   
3   31      M  United Kingdom   NaN           NaN            Yes       Yes   
4   31      M   United States    TX           NaN             No        No   

  work_interfere    no_employees remote_work  ...   anonymity  \
0          Often             NaN          No  ...         Yes   
1         Rarely  More than 1000          No  ...  Don't know   
2         Rarely             NaN          No  ...  Don't know   
3          Often          26-100          No  ...          No   
4          Never         100-500         Yes  ...  Don't know   

                leave mental_health_consequence phys_health_consequence  \
0       Somewhat easy                        No                      No   
1          Don't know                     Maybe                      No   
2  Somewhat difficult                        No                      No   
3  Somewhat difficult                       Yes                     Yes   
4          Don't know                        No                      No   

      coworkers supervisor mental_health_interview phys_health_interview  \
0  Some of them        Yes                      No                 Maybe   
1            No         No                      No                    No   
2           Yes        Yes                     Yes                   Yes   
3  Some of them         No                   Maybe                 Maybe   
4  Some of them        Yes                     Yes                   Yes   

  mental_vs_physical obs_consequence  
0                Yes              No  
1         Don't know              No  
2                 No              No  
3                 No             Yes  
4         Don't know              No  

[5 rows x 25 columns]

### Automatic Assessment

- head and tail
- sample
- info
- isnull
- duplicated
- describe

In [ ]:
df.head()

Age Gender         Country state self_employed family_history treatment  \
0   37      F   United States    IL           NaN             No       Yes   
1   44      M   United States    IN           NaN             No        No   
2   32      M          Canada   NaN           NaN             No        No   
3   31      M  United Kingdom   NaN           NaN            Yes       Yes   
4   31      M   United States    TX           NaN             No        No   

  work_interfere    no_employees remote_work  ...   anonymity  \
0          Often             NaN          No  ...         Yes   
1         Rarely  More than 1000          No  ...  Don't know   
2         Rarely             NaN          No  ...  Don't know   
3          Often          26-100          No  ...          No   
4          Never         100-500         Yes  ...  Don't know   

                leave mental_health_consequence phys_health_consequence  \
0       Somewhat easy                        No                      No   
1          Don't know                     Maybe                      No   
2  Somewhat difficult                        No                      No   
3  Somewhat difficult                       Yes                     Yes   
4          Don't know                        No                      No   

      coworkers supervisor mental_health_interview phys_health_interview  \
0  Some of them        Yes                      No                 Maybe   
1            No         No                      No                    No   
2           Yes        Yes                     Yes                   Yes   
3  Some of them         No                   Maybe                 Maybe   
4  Some of them        Yes                     Yes                   Yes   

  mental_vs_physical obs_consequence  
0                Yes              No  
1         Don't know              No  
2                 No              No  
3                 No             Yes  
4         Don't know              No  

[5 rows x 25 columns]

In [ ]:
# MAKING AGE OF POSITIVE
df['Age'] = df['Age'].abs()

In [ ]:
df.shape

(1259, 25)

In [ ]:
df.columns

Index(['Age', 'Gender', 'Country', 'state', 'self_employed', 'family_history',
       'treatment', 'work_interfere', 'no_employees', 'remote_work',
       'tech_company', 'benefits', 'care_options', 'wellness_program',
       'seek_help', 'anonymity', 'leave', 'mental_health_consequence',
       'phys_health_consequence', 'coworkers', 'supervisor',
       'mental_health_interview', 'phys_health_interview',
       'mental_vs_physical', 'obs_consequence'],
      dtype='object')

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1259 entries, 0 to 1258
Data columns (total 25 columns):
 #   Column                     Non-Null Count  Dtype 
---  ------                     --------------  ----- 
 0   Age                        1259 non-null   int64 
 1   Gender                     1259 non-null   object
 2   Country                    1259 non-null   object
 3   state                      744 non-null    object
 4   self_employed              1241 non-null   object
 5   family_history             1259 non-null   object
 6   treatment                  1259 non-null   object
 7   work_interfere             995 non-null    object
 8   no_employees               807 non-null    object
 9   remote_work                1259 non-null   object
 10  tech_company               1259 non-null   object
 11  benefits                   1259 non-null   object
 12  care_options               1259 non-null   object
 13  wellness_program           1259 non-null   object
 14  seek_hel

In [ ]:
df.describe()

Age
count  1259.000000
mean     32.012708
std       7.370601
min       5.000000
25%      27.000000
50%      31.000000
75%      36.000000
max      72.000000

In [ ]:
#DISPLAY THE COLUMNS WHERE STATE IS NULL
df[df['state'].isnull()]

Age Gender         Country state self_employed family_history treatment  \
2      32      M          Canada   NaN           NaN             No        No   
3      31      M  United Kingdom   NaN           NaN            Yes       Yes   
7      39      M          Canada   NaN           NaN             No        No   
9      23      M          Canada   NaN           NaN             No        No   
11     29      M        Bulgaria   NaN           NaN             No        No   
...   ...    ...             ...   ...           ...            ...       ...   
1244   32      F  United Kingdom   NaN            No             No        No   
1245   22      M       Australia   NaN            No            Yes       Yes   
1247   36      M         Finland   NaN            No             No       Yes   
1251   36      M    South Africa   NaN            No            Yes       Yes   
1254   26      M  United Kingdom   NaN            No             No       Yes   

     work_interfere    no_employees remote_work  ...   anonymity  \
2            Rarely             NaN          No  ...  Don't know   
3             Often          26-100          No  ...          No   
7             Never             NaN         Yes  ...         Yes   
9             Never          26-100          No  ...  Don't know   
11            Never         100-500         Yes  ...  Don't know   
...             ...             ...         ...  ...         ...   
1244            NaN  More than 1000          No  ...  Don't know   
1245          Often         100-500          No  ...         Yes   
1247          Often             NaN          No  ...          No   
1251          Often         100-500          No  ...  Don't know   
1254            NaN          26-100          No  ...  Don't know   

                   leave mental_health_consequence phys_health_consequence  \
2     Somewhat difficult                        No                      No   
3     Somewhat difficult                       Yes                     Yes   
7             Don't know                        No                      No   
9             Don't know                        No                      No   
11            Don't know                        No                      No   
...                  ...                       ...                     ...   
1244          Don't know                     Maybe                      No   
1245          Don't know                     Maybe                   Maybe   
1247      Very difficult                       Yes                      No   
1251       Somewhat easy                        No                      No   
1254       Somewhat easy                        No                      No   

         coworkers    supervisor mental_health_interview  \
2              Yes           Yes                     Yes   
3     Some of them            No                   Maybe   
7               No            No                      No   
9              Yes           Yes                   Maybe   
11             Yes           Yes                     Yes   
...            ...           ...                     ...   
1244  Some of them           Yes                      No   
1245            No           Yes                      No   
1247  Some of them            No                      No   
1251  Some of them           Yes                      No   
1254  Some of them  Some of them                      No   

     phys_health_interview mental_vs_physical obs_consequence  
2                      Yes                 No              No  
3                    Maybe                 No             Yes  
7                       No                 No              No  
9                    Maybe                Yes              No  
11                     Yes         Don't know              No  
...                    ...                ...             ...  
1244                   Yes                 No              No  
1245                 Maybe         Don't know 

515 out of 1259 values (40.84 %) are missing for state column, there is no way to find out the information of the state for an individual, dropping the `state` column

In [ ]:
#DROPPING STATE COLUMN
df = df.drop(columns=['state'])

In [ ]:
df.head()

Age Gender         Country self_employed family_history treatment  \
0   37      F   United States           NaN             No       Yes   
1   44      M   United States           NaN             No        No   
2   32      M          Canada           NaN             No        No   
3   31      M  United Kingdom           NaN            Yes       Yes   
4   31      M   United States           NaN             No        No   

  work_interfere    no_employees remote_work tech_company  ...   anonymity  \
0          Often             NaN          No          Yes  ...         Yes   
1         Rarely  More than 1000          No           No  ...  Don't know   
2         Rarely             NaN          No          Yes  ...  Don't know   
3          Often          26-100          No          Yes  ...          No   
4          Never         100-500         Yes          Yes  ...  Don't know   

                leave mental_health_consequence phys_health_consequence  \
0       Somewhat easy                        No                      No   
1          Don't know                     Maybe                      No   
2  Somewhat difficult                        No                      No   
3  Somewhat difficult                       Yes                     Yes   
4          Don't know                        No                      No   

      coworkers supervisor mental_health_interview phys_health_interview  \
0  Some of them        Yes                      No                 Maybe   
1            No         No                      No                    No   
2           Yes        Yes                     Yes                   Yes   
3  Some of them         No                   Maybe                 Maybe   
4  Some of them        Yes                     Yes                   Yes   

  mental_vs_physical obs_consequence  
0                Yes              No  
1         Don't know              No  
2                 No              No  
3                 No             Yes  
4         Don't know              No  

[5 rows x 24 columns]

In [ ]:
#DISPLAY THE COLUMNS WHERE SELF EMPLOYED IS NULL
df[df['self_employed'].isnull()]

Age Gender         Country self_employed family_history treatment  \
0    37      F   United States           NaN             No       Yes   
1    44      M   United States           NaN             No        No   
2    32      M          Canada           NaN             No        No   
3    31      M  United Kingdom           NaN            Yes       Yes   
4    31      M   United States           NaN             No        No   
5    33      M   United States           NaN            Yes        No   
6    35      F   United States           NaN            Yes       Yes   
7    39      M          Canada           NaN             No        No   
8    42      F   United States           NaN            Yes       Yes   
9    23      M          Canada           NaN             No        No   
10   31      M   United States           NaN             No       Yes   
11   29      M        Bulgaria           NaN             No        No   
12   42      F   United States           NaN            Yes       Yes   
13   36      M   United States           NaN            Yes        No   
14   27      M          Canada           NaN             No        No   
15   29      F   United States           NaN            Yes       Yes   
16   23      M  United Kingdom           NaN             No       Yes   
17   32      M   United States           NaN             No       Yes   

   work_interfere    no_employees remote_work tech_company  ...   anonymity  \
0           Often             NaN          No          Yes  ...         Yes   
1          Rarely  More than 1000          No           No  ...  Don't know   
2          Rarely             NaN          No          Yes  ...  Don't know   
3           Often          26-100          No          Yes  ...          No   
4           Never         100-500         Yes          Yes  ...  Don't know   
5       Sometimes             NaN          No          Yes  ...  Don't know   
6       Sometimes             NaN         Yes          Yes  ...          No   
7           Never             NaN         Yes          Yes  ...         Yes   
8       Sometimes         100-500          No          Yes  ...          No   
9           Never          26-100          No          Yes  ...  Don't know   
10      Sometimes             NaN         Yes          Yes  ...  Don't know   
11          Never         100-500         Yes          Yes  ...  Don't know   
12      Sometimes          26-100          No           No  ...  Don't know   
13          Never        500-1000          No          Yes  ...  Don't know   
14          Never             NaN          No          Yes  ...  Don't know   
15         Rarely          26-100          No          Yes  ...  Don't know   
16      Sometimes          26-100         Yes          Yes  ...  Don't know   
17      Sometimes             NaN          No          Yes  ...  Don't know   

                 leave mental_health_consequence phys_health_consequence  \
0        Somewhat easy                        No                      No   
1           Don't know                     Maybe                      No   
2   Somewhat difficult                        No                      No   
3   Somewhat difficult                       Yes                     Yes   
4           Don't know                        No                      No   
5           Don't know                        No                      No   
6   Somewhat difficult                     Maybe                   Maybe   
7           Don't know                        No                      No   
8       Very difficult                     Maybe                      No   
9           Don't know                        No                      No   
10          Don't know                        No                      No   
11          Don't know                        No                      No   
12  Somewhat difficult                       Yes                     Yes   
13          Don't know                        No  

In [ ]:
#SELF_EMPLOYED IS SET TO YES WHEREVER THERE IS NAN
df['self_employed'].fillna(value='Yes', inplace=True)

In [ ]:
#DISPLAY THE COLUMNS WHERE WORK INTERFERE IS NULL
df[df['work_interfere'].isnull()]

Age Gender         Country self_employed family_history treatment  \
19     36      M          France           Yes            Yes        No   
26     33      M   United States            No             No        No   
37     38      M        Portugal            No             No        No   
38     50      M   United States            No             No        No   
41     35      M   United States            No             No        No   
...   ...    ...             ...           ...            ...       ...   
1226   39      M     Netherlands            No             No        No   
1229   39      M          Greece            No             No        No   
1244   32      F  United Kingdom            No             No        No   
1254   26      M  United Kingdom            No             No       Yes   
1257   46      F   United States            No             No        No   

     work_interfere    no_employees remote_work tech_company  ...   anonymity  \
19              NaN             NaN         Yes          Yes  ...         Yes   
26              NaN             NaN          No          Yes  ...  Don't know   
37              NaN         100-500          No          Yes  ...  Don't know   
38              NaN         100-500          No          Yes  ...  Don't know   
41              NaN  More than 1000         Yes          Yes  ...  Don't know   
...             ...             ...         ...          ...  ...         ...   
1226            NaN  More than 1000          No           No  ...  Don't know   
1229            NaN             NaN         Yes          Yes  ...         Yes   
1244            NaN  More than 1000          No           No  ...  Don't know   
1254            NaN          26-100          No          Yes  ...  Don't know   
1257            NaN         100-500         Yes          Yes  ...  Don't know   

                   leave mental_health_consequence phys_health_consequence  \
19         Somewhat easy                        No                      No   
26            Don't know                     Maybe                   Maybe   
37         Somewhat easy                     Maybe                      No   
38            Don't know                        No                      No   
41    Somewhat difficult                       Yes                     Yes   
...                  ...                       ...                     ...   
1226       Somewhat easy                     Maybe                   Maybe   
1229          Don't know                       Yes                      No   
1244          Don't know                     Maybe                      No   
1254       Somewhat easy                        No                      No   
1257          Don't know                       Yes                      No   

         coworkers    supervisor mental_health_interview  \
19    Some of them  Some of them                   Maybe   
26    Some of them            No                      No   
37    Some of them  Some of them                      No   
38    Some of them           Yes                      No   
41    Some of them            No                      No   
...            ...           ...                     ...   
1226            No  Some of them                      No   
1229            No            No                      No   
1244  Some of them           Yes                      No   
1254  Some of them  Some of them                      No   
1257            No            No                      No   

     phys_health_interview mental_vs_physical obs_consequence  
19                   Maybe         Don't know              No  
26                      No         Don't know              No  
37                   Maybe                 No              No  
38                   Maybe         Don't know              No  
41                   Maybe         Don't know              No  
...                    ...                ...             ...  
1226                    No

264 People have not mentioned whether their mental health problem is due to work interference or not. So we are filling it with sometimes

In [ ]:
df['work_interfere'].fillna(value='Sometimes', inplace=True)

In [ ]:
df['work_interfere'].value_counts()

Sometimes    729
Never        213
Rarely       173
Often        144
Name: work_interfere, dtype: int64

no_employees is a column which is difficult to deal with, it's inconsistent and 64.15% of values are missing. Dropping this column.

In [ ]:
df = df.drop(columns = ['no_employees'])

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1259 entries, 0 to 1258
Data columns (total 23 columns):
 #   Column                     Non-Null Count  Dtype 
---  ------                     --------------  ----- 
 0   Age                        1259 non-null   int64 
 1   Gender                     1259 non-null   object
 2   Country                    1259 non-null   object
 3   self_employed              1259 non-null   object
 4   family_history             1259 non-null   object
 5   treatment                  1259 non-null   object
 6   work_interfere             1259 non-null   object
 7   remote_work                1259 non-null   object
 8   tech_company               1259 non-null   object
 9   benefits                   1259 non-null   object
 10  care_options               1259 non-null   object
 11  wellness_program           1259 non-null   object
 12  seek_help                  1259 non-null   object
 13  anonymity                  1259 non-null   object
 14  leave   

In [ ]:
# DEALING WITH GENDER COLUMN WHERE THE VALUES ARE INCONSISTENT
df['Gender'] = df['Gender'].replace(['M ','m'], 'M')

In [ ]:
df['Gender'] = df['Gender'].replace(['F ','f','Make'], 'F')

In [ ]:
df['Gender'].value_counts()

M    1003
F     256
Name: Gender, dtype: int64

In [ ]:
df['Country'].value_counts()

United States             751
United Kingdom            185
Canada                     72
Germany                    45
Ireland                    27
Netherlands                27
Australia                  21
France                     13
India                      10
New Zealand                 8
Poland                      7
Switzerland                 7
Sweden                      7
Italy                       7
South Africa                6
Belgium                     6
Brazil                      6
Israel                      5
Singapore                   4
Bulgaria                    4
Austria                     3
Finland                     3
Mexico                      3
Russia                      3
Denmark                     2
Greece                      2
Colombia                    2
Croatia                     2
Portugal                    2
Moldova                     1
Georgia                     1
Bahamas, The                1
China                       1
Thailand  

In [ ]:
df['family_history'].value_counts()

No     767
Yes    492
Name: family_history, dtype: int64

In [ ]:
df['treatment'].value_counts()

Yes    637
No     622
Name: treatment, dtype: int64

In [ ]:
df['work_interfere'].value_counts()

Sometimes    729
Never        213
Rarely       173
Often        144
Name: work_interfere, dtype: int64

In [ ]:
df['remote_work'].value_counts()

No     883
Yes    376
Name: remote_work, dtype: int64

In [ ]:
df['tech_company'].value_counts()

Yes    1031
No      228
Name: tech_company, dtype: int64

In [ ]:
df['benefits'].value_counts()

Yes           477
Don't know    408
No            374
Name: benefits, dtype: int64

In [ ]:
df['care_options'].value_counts()

No          501
Yes         444
Not sure    314
Name: care_options, dtype: int64

In [ ]:
df['wellness_program'].value_counts()

No            842
Yes           229
Don't know    188
Name: wellness_program, dtype: int64

In [ ]:
df['seek_help'].value_counts()

No            646
Don't know    363
Yes           250
Name: seek_help, dtype: int64

In [ ]:
df['anonymity'].value_counts()

Don't know    819
Yes           375
No             65
Name: anonymity, dtype: int64

In [ ]:
df['leave'].value_counts()

Don't know            563
Somewhat easy         266
Very easy             206
Somewhat difficult    126
Very difficult         98
Name: leave, dtype: int64

In [ ]:
df['mental_health_consequence'].value_counts()

No       490
Maybe    477
Yes      292
Name: mental_health_consequence, dtype: int64

In [ ]:
df['phys_health_consequence'].value_counts()

No       925
Maybe    273
Yes       61
Name: phys_health_consequence, dtype: int64

In [ ]:
df.columns

Index(['Age', 'Gender', 'Country', 'self_employed', 'family_history',
       'treatment', 'work_interfere', 'remote_work', 'tech_company',
       'benefits', 'care_options', 'wellness_program', 'seek_help',
       'anonymity', 'leave', 'mental_health_consequence',
       'phys_health_consequence', 'coworkers', 'supervisor',
       'mental_health_interview', 'phys_health_interview',
       'mental_vs_physical', 'obs_consequence'],
      dtype='object')

In [ ]:
df['coworkers'].value_counts()

Some of them    774
No              260
Yes             225
Name: coworkers, dtype: int64

In [ ]:
df['supervisor'].value_counts()

Yes             516
No              393
Some of them    350
Name: supervisor, dtype: int64

In [ ]:
df['mental_health_interview'].value_counts()

No       1008
Maybe     207
Yes        44
Name: mental_health_interview, dtype: int64

In [ ]:
df['phys_health_interview'].value_counts()

Maybe    557
No       500
Yes      202
Name: phys_health_interview, dtype: int64

In [ ]:
df['mental_vs_physical'].value_counts()

Don't know    576
Yes           343
No            340
Name: mental_vs_physical, dtype: int64

In [ ]:
df['obs_consequence'].value_counts()

No     1075
Yes     184
Name: obs_consequence, dtype: int64

In [ ]:
df.to_csv('final_data.csv', index = False)